### ISSUE TITLE
Istio no longer supporting ServiceEntry with single IP address and CIDR range

### ISSUE THREAD

**Issue by alexandermarston** [istio/istio#52944]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

In Istio versions prior to 1.23, we were able to create a ServiceEntry that had multiple IP Addresses including a single IP Address as well as CIDR ranges - it appears in 1.23 this is no longer possible and results in the following error:

```
2024-08-30T10:47:52.555598Z	warn	delta	ADS:LDS: ACK ERROR hello-world-569c66b9dd-v5jvj.production-3 Internal:Error adding/updating listener(s) 89.143.62.178_7896: malformed IP address: 203.45.160.0/20
```

At this point, my Istio Sidecars are unable to process anything:

```
2024-08-30T10:47:58.656991Z     warn    Envoy proxy is NOT ready: config received from XDS server, but was rejected: cds updates: 2 successful, 0 rejected; lds updates: 0 successful, 2 rejected
```

To re-produce create the following ServiceEntry:

```
apiVersion: networking.istio.io/v1
kind: ServiceEntry
metadata:
  name: my-first-test-host
  namespace: istio-system
spec:
  hosts:
    - "afailing.serviceentry.com"
  addresses:
    - "89.143.62.178"
    - "203.45.160.0/20"
  ports:
    - number: 7896
      name: https
      protocol: HTTPS
  resolution: NONE
  location: MESH_EXTERNAL
```

Reconciliation does work if I split the ServiceEntry into two separate ServiceEntries, like so:

```
apiVersion: networking.istio.io/v1
kind: ServiceEntry
metadata:
  name: my-first-test-host-1
  namespace: istio-system
spec:
  hosts:
    - "afailing.serviceentry.com"
  addresses:
    - "89.143.62.178"
  ports:
    - number: 7896
      name: https
      protocol: HTTPS
  resolution: NONE
  location: MESH_EXTERNAL
---
apiVersion: networking.istio.io/v1
kind: ServiceEntry
metadata:
  name: my-first-test-host-2
  namespace: istio-system
spec:
  hosts:
    - "afailing.serviceentry.com"
  addresses:
    - "203.45.160.0/20"
  ports:
    - number: 7896
      name: https
      protocol: HTTPS
  resolution: NONE
  location: MESH_EXTERNAL
```

### Version

```Text
➜  ~ kubectl version
Client Version: v1.29.3
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.30.3-eks-2f46c53

➜  ~ helm version --short
v3.12.3+g3a31588

➜  ~ helm list
NAME      	NAMESPACE   	REVISION	UPDATED                                	STATUS  	CHART        	APP VERSION
istio-base	istio-system	10      	2024-08-30 10:41:09.375457406 +0000 UTC	deployed	base-1.23.0  	1.23.0
istiod    	istio-system	20      	2024-08-30 09:36:44.584703674 +0000 UTC	deployed	istiod-1.23.0	1.23.0
```

**Comment by howardjohn** [comment]
Likely the change that triggered this was https://github.com/istio/istio/pull/51967

**Comment by howardjohn** [comment]
I've confirmed this is fixed now that https://github.com/istio/istio/pull/51967 is reverted on older branches and forward-fixed on master. Thanks!

**Comment by AgustinICAI** [comment]
Hi, why this has not been solved for istio 1.24.2 or 1.24.3? I can still find the old code that was reverted
